### ISSUE TITLE
Missing warning when using ClassVar without subscript

### ISSUE THREAD

**Issue by lkct** [microsoft/pyright#11154]
**Describe the bug**
Please see code below. When bare ClassVar is used, instance assignment is not warned.
As far as I understand the [specs](https://typing.python.org/en/latest/spec/class-compat.html#classvar), using ClassVar without a subscript type only changes how its type is inferred, but not its behavior as a class variable.

**Code or Screenshots**

Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCANgIYDOZAaiSALABQDDAxqRVAELkCmAXA1EFRgYML0JsqNANqoYAXSgBeKAAYBQgEY1xxclJDK1TRvVb6oAZQCumgBRcy3AJT96QqABNuwKAH0-VCQYALsnImBnKABaAD4oADkwFD4ND0Fw4AA6ETAjVUEAYigQbhw8AEEYGBAkTWsYbgrmZm4KAEkKa24ocihuAA8EbmZGzzT0zKztQxUCqGKyAAswayJPKCWSADcemCWeshIIHqQunvr8FDB8AHcaFPH6IA)

```python
from typing import ClassVar


class Base:
    foo: ClassVar[int] = 0
    bar: ClassVar = 0


class Sub(Base):
    def __init__(self) -> None:
        self.foo = 0  # reportAttributeAccessIssue as expected
        self.bar = 0  # should have the same issue but not warned

```

**VS Code extension or command-line**
VSCode with Pylance 2025.10.4, as well as pyright playground with 1.1.407
